app/controllers/pages_controller.rb

```diff
@@ -5,11 +5,13 @@ def home
       @timeline = true
 
       if params[:before]
-        @before = Nweet.find_by(url_digest: params[:before])
-        @feed_items = current_user.timeline.where('did_at < ?', @before.did_at).limit(10)
-        render partial: 'nweets/nweet', collection: @feed_items
+        date = Time.at(params[:before].to_i)
+        @feed_items = current_user.timeline.where('did_at < ?', date).limit(10)
+        @before = @feed_items.last&.did_at&.to_i
+        render partial: 'nweets/nweets'
       else
         @feed_items = current_user.timeline.limit(10)
+        @before = @feed_items.last&.did_at&.to_i
```

Comment:
若干重複があって嫌ですが、@beforeの定義の部分だけのためにif文2つに変えるのも違う気がして難しいですね……。

Reply:
ちょっと試してみたらifの外に書いても同じ動きをするっぽい..?
ちょっとわかりづらくなっている気もしなくはないけど
```suggestion
        @feed_items = current_user.timeline.where('did_at < ?', date).limit(10)
        render partial: 'nweets/nweets'
      else
        @feed_items = current_user.timeline.limit(10)
      end
      @before = @feed_items.last&.did_at&.to_i
```

Reply:
あー……これなら元の同じコードでいい気がしますね（動くんだ……）